ledger/participant-integration-api/src/main/scala/platform/store/cache/InMemoryFanoutBuffer.scala

```diff
@@ -104,26 +117,77 @@ class EventsBuffer[ENTRY](
       },
     )
 
-  /** Removes entries starting from the buffer tail up until `endInclusive`.
+  /** Lookup the accepted transaction update by transaction id. */
+  def lookup(transactionId: TransactionId): Option[TransactionLogUpdate.TransactionAccepted] =
+    _lookupMap.get(transactionId)
+
+  /** Removes entries starting from the buffer head up until `endInclusive`.
     *
     * @param endInclusive The last inclusive (highest) buffer offset to be pruned.
     */
   def prune(endInclusive: Offset): Unit =
     Timed.value(
       pruneTimer,
       synchronized {
-        _bufferLog = _bufferLog.view.map(_._1).search(endInclusive) match {
-          case Found(foundIndex) => _bufferLog.drop(foundIndex + 1)
-          case InsertionPoint(insertionPoint) => _bufferLog.drop(insertionPoint)
+        val dropCount = _bufferLog.view.map(_._1).search(endInclusive) match {
+          case Found(foundIndex) => foundIndex + 1
+          case InsertionPoint(insertionPoint) => insertionPoint
         }
+
+        dropOldest(dropCount)
       },
     )
 
   /** Remove all buffered entries */
-  def flush(): Unit = synchronized { _bufferLog = Vector.empty }
+  def flush(): Unit = synchronized {
+    _bufferLog = Vector.empty
+    _lookupMap = Map.empty
+  }
+
+  private def ensureSize(targetSize: Int): Unit = synchronized {
+    val currentBufferLogSize = _bufferLog.size
+    val currentLookupMapSize = _lookupMap.size
+
+    if (currentLookupMapSize <= currentBufferLogSize) {
+      bufferSizeHistogram.update(currentBufferLogSize)
+
+      if (currentBufferLogSize > targetSize) {
+        dropOldest(dropCount = currentBufferLogSize - targetSize)
+      }
+    } else {
+      // This is an error condition. If encountered, clear the in-memory fan-out buffers.
+      ContextualizedLogger
```

Comment:
For some error conditions we throw exceptions, here we log, flush and continue.  Would a consistent approach be better or does situation determine the remedy.

Reply:
I usually think that if we can be less disruptive to clients, we should be (if the encountered error is benign). I prefer to throw where we don't know how to recover, but here the situation is manageable. Please do note that this guard is only there because we are not explicitly enforcing the size requirement on the lookupMap, but on the bufferLog vector.